model.py

```diff
@@ -6558,6 +6558,11 @@ class Credential(Base):
     credential = Column(String)
     expires = Column(DateTime)
 
+    # One Credential can have many associated DRMDeviceIdentifiers.
+    drm_device_identifiers = relationship(
+        "DRMDeviceIdentifier", backref=backref("credential", lazy='joined')
+    )
+    
```

Comment:
I associate DRMDeviceIdentifier with Credential rather than Patron because your Adobe device IDs are associated with your Adobe ID, not your library account. I want to support tracking DRM device identifiers for (let's say) a patron's accounts with different URMS stores. This requires that one Patron keep multiple lists of DRMDeviceIdentifiers, one for each DRM-related Credential.

Although the Adobe ID itself is in DelegatedPatronIdentifier, I associate DRMDeviceIdentifier with Credential rather than with DelegatedPatronIdentifier. That's because I want each library to keep track of the DRM device IDs of its own patrons, and most libraries won't have access to DelegatedPatronIdentifiers.